lib/pgBackRest/Common/Io/Base.pm

```diff
@@ -0,0 +1,126 @@
+####################################################################################################################################
+# Base object for IO/filter objects.
+####################################################################################################################################
+package pgBackRest::Common::Io::Base;
+
+use strict;
+use warnings FATAL => qw(all);
+use Carp qw(confess);
+use English '-no_match_vars';
+
+use Exporter qw(import);
+    our @EXPORT = qw();
+use Scalar::Util qw(blessed);
+
+use pgBackRest::Common::Log;
+
+####################################################################################################################################
+# Package name constant
+####################################################################################################################################
+use constant COMMON_IO_BASE                                         => __PACKAGE__;
+    push @EXPORT, qw(COMMON_IO_BASE);
+
+####################################################################################################################################
+# Default buffer max
+####################################################################################################################################
+use constant COMMON_IO_BUFFER_MAX                                   => 4194304;
```

Comment:
CSHANG Shouldn't this always be the same as OPTION_DEFAULT_BUFFER_SIZE? And visa-versa - so could we have a common file where these are set so they don't get out of sync?